tests/ci/digest_helper.py

```diff
@@ -23,46 +23,43 @@ def _digest_file(file: Path, hash_object: HASH) -> None:
             hash_object.update(chunk)
 
 
-def _digest_directory(directory: Path, hash_object: HASH) -> None:
-    assert directory.is_dir()
-    for p in sorted(directory.rglob("*")):
-        if p.is_symlink() and p.is_dir():
-            # The symlink directory is not listed recursively, so we process it manually
-            (_digest_directory(p, hash_object))
-        if p.is_file():
-            (_digest_file(p, hash_object))
-
-
-def digest_path(path: Path, hash_object: Optional[HASH] = None) -> HASH:
+def digest_path(
+    path: Union[Path, str],
+    hash_object: Optional[HASH] = None,
+    exclude_files: Optional[Iterable[str]] = None,
+) -> HASH:
     """Calculates md5 (or updates existing hash_object) hash of the path, either it's
     directory or file"""
+    if isinstance(path, str):
+        path = Path(path)
     hash_object = hash_object or md5()
-    if path.is_dir():
-        _digest_directory(path, hash_object)
-    elif path.is_file():
-        _digest_file(path, hash_object)
+    # FIXME: what do we do with links?
+    if path.is_file():
+        if not exclude_files or not any(path.name.endswith(x) for x in exclude_files):
+            _digest_file(path, hash_object)
+    elif path.is_dir():
+        for p in sorted(path.iterdir()):
+            digest_path(p, hash_object, exclude_files)
+    else:
+        pass  # broken symlink
     return hash_object
 
 
-def digest_paths(paths: Iterable[Path], hash_object: Optional[HASH] = None) -> HASH:
+def digest_paths(
+    paths: Iterable[Path],
+    hash_object: Optional[HASH] = None,
+    exclude_files: Optional[Iterable[str]] = None,
+) -> HASH:
     """Calculates aggregated md5 (or updates existing hash_object) hash of passed paths.
     The order is processed as given"""
     hash_object = hash_object or md5()
+    paths = sorted(p.absolute() for p in paths)
     for path in paths:
         if path.exists():
-            digest_path(path, hash_object)
+            digest_path(path, hash_object, exclude_files)
     return hash_object
 
 
-def digest_consistent_paths(
```

Comment:
It's questionable, but ok, let's get rid of it. Although, this is theoretically a liiiiiittle bit slower.

Reply:
if someone needs inconsistent hash when we can create function digest_inconsistent_paths() :) 
but for me it looks a bit weird as we do not sort dirs on first layer but on next layers we start sorting them in _digest_directory()